src/BulkIssueCreatorCLI/Program.cs

```diff
@@ -0,0 +1,355 @@
+                        SaveIssuesToProgressFile(progressFile, parsedIssueData);
+                    }
+                    else
+                    {
+                        Colorizer.WriteLine("[Red!Error]: Could not create issue [Yellow!{0}]. [Red!Stopping]", issue.Title);
+                    }
+                    await Task.Delay(500);
+                }
+
+
+            }
+
+            //// create all the issues, without being in an epic.
+            //foreach (IssueToCreateWithEpic issue in parsedIssueData)
+            //{
+            //    Colorizer.WriteLine("Create [Yellow!{0}]? [Green!y]/[Red!n]", issue.Title);
+            //    if (Console.ReadKey().Key != ConsoleKey.Y)
+            //    {
+            //        continue;
+            //    }
+
+            //    using (LoggingScope ls = new LoggingScope("Creating [Yellow!{0}]", issue.Title))
+            //    {
+            //        if (!string.IsNullOrEmpty(issue.EpicTitle))
+            //        {
+            //            issue.Epic = await GetEpicFromTitle(issue);
+            //        }
+
+            //        (bool issueCreated, string errorOrUrl) = await s_issueManager.TryCreateNewIssueAsync(issue, false);
+
+            //        if (issueCreated == true)
+            //        {
+            //            Colorizer.WriteLine("Issue [Yellow!{0}] created with url [Cyan!{1}]", issue.Title, errorOrUrl);
+            //        }
+            //        else
+            //        {
+            //            Colorizer.WriteLine("[Red!Error]: Could not create issue [Yellow!{0}]. [Red!Stopping]", issue.Title);
+            //        }
+            //        await Task.Delay(500);
+            //    }
+            //}
+        }
+
+        private static async Task<IssueDescription> GetEpicFromTitle(IssueToCreateWithEpic issue)
+        {
+            List<IssueDescription> epics = await s_issueManager.GetEpicsWithTitleAsync(issue.EpicTitle, $"{issue.EpicOrg}\\{issue.EpicRepo}");
+            await Task.Delay(500);
+
+            Colorizer.WriteLine("Found parent epic [Cyan!{0}].", epics.FirstOrDefault().Issue.Title);
+
+            return epics.FirstOrDefault();
+        }
+
+        private static async Task ValidateAndSetMilestonesAsync(List<IssueToCreateWithEpic> parsedIssueData)
+        {
+            foreach (IssueToCreateWithEpic issue in parsedIssueData)
+            {
+                // get the milestone
+                if (string.IsNullOrEmpty(issue.MilestoneText))
+                {
+                    continue;
+                }
+
+                issue.Milestone = await s_issueManager.GetMilestoneAsync(issue.Organization, issue.Repository, issue.MilestoneText);
+                if (issue.Milestone == null)
+                {
+                    Colorizer.WriteLine("[Red!Error] Could not find milestone [Yellow!{0}] in repository [Yellow!{1}\\{2}]", issue.MilestoneText, issue.Organization, issue.Repository);
```

Comment:
Milestones against their graphql endpoints aren't case-sensitive, but labels are. I opened an issue a long while ago and even submitted a PR to their `gh` CLI that was eventually closed since they said they were going to tackle it on the backend, but I've seen no indication this is fixed. If/when you deal with labels (the UI tool does), something to keep in mind: maybe suggest to the user they make sure they use the same case.

Reply:
That is a great point! One thing that I can do is actually validate that the labels specified exist and bail in case they don't. That way you don't get into a half-baked state where some issues are created and some aren't.

Reply:
https://github.com/Azure/azure-sdk-tools/issues/1673

Reply:
Opened #40.